system/mp/mpcomm.cpp

```diff
@@ -1884,8 +1889,17 @@ class CMPPacketReader: public ISocketSelectNotify, public CInterface
                         // TBD IPV6 here
                         SocketEndpoint ep;
                         sock->getPeerEndpoint(ep);
-                        IMP_Exception *e=new CMPException(MPERR_protocol_version_mismatch,ep);
-                        throw e;
+                        if (periodicTimer.hasElapsed())
+                        {
+                            StringBuffer packetHdrBytes("Packet Header bytes: ");
+                            hexdump2string((byte const *)&hdr, sizeof(hdr), packetHdrBytes);
```

Comment:
Should this be std::min(packetHdrBytes, sizeRead)?

Also, should the number of bytes be logged in the message?

Reply:
>Also, should the number of bytes be logged in the message?

could make hexdump2string prefix message with # bytes..

Reply:
have changed hexdump2string to include # bytes